Framework/API/src/NearestNeighbours.cpp

```diff
@@ -37,9 +23,9 @@ NearestNeighbours::NearestNeighbours(
  * be ignored.
  */
 NearestNeighbours::NearestNeighbours(
-    int nNeighbours, boost::shared_ptr<const Instrument> instrument,
-    const ISpectrumDetectorMapping &spectraMap, bool ignoreMaskedDetectors)
-    : m_instrument(instrument), m_spectraMap(spectraMap),
+    int nNeighbours, const SpectrumInfo &spectrumInfo,
+    const std::vector<specnum_t> spectrumNumbers, bool ignoreMaskedDetectors)
+    : m_spectrumInfo(spectrumInfo), m_spectrumNumbers(spectrumNumbers),
```

Comment:
@SimonHeybrock I assume thought has been given to lifetimes of `NearestNeighbour` objects as compared to `SpectrumInfos` and whether this can be abused in a way that would result in dead references.